numpy/lib/npyio.py

```diff
@@ -259,6 +270,22 @@ def __getitem__(self, key):
         else:
             raise KeyError("%s is not a file in the archive" % key)
 
+    def __repr__(self):
+        # Get filename or default to `object`
+        if isinstance(self.fid, str):
+            filename = self.fid
+        else:
+            filename = getattr(self.fid, "name", "object")
+
+        # Get the name of arrays
+        if len(self.files) > self.MAX_REPR_ARRAY_COUNT:
+            array_names = "%s..." % (
+                ', '.join(self.files[:self.MAX_REPR_ARRAY_COUNT])
+            )
+        else:
+            array_names = ', '.join(self.files)
+        return "<NpzFile %s containing %s>" % (filename, array_names)
```

Comment:
```suggestion
        return f"<NpzFile {filename!r} containing {array_names}"
```
f-strings are nice, and maybe `repr` is nicer for the filename (mainly adds a `'`)?

Reply:
Just to add my own little nits here, I very slightly prefer replacing "containing" with "with keys" or "with files" - mostly because that gives a better hint how to get the full listing.

Reply:
Thanks! Both make this cleaner. Added via [3f4b84d](https://github.com/numpy/numpy/pull/23357/commits/3f4b84d24e6c69371a9582326dd2238200a27b52)